controllers/usersignupcleanup/usersignup_cleanup_controller.go

```diff
@@ -82,9 +82,11 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	}
 	reqLogger = reqLogger.WithValues("username", instance.Spec.Username)
 
+	activations, activationsAnnotationPresent := instance.Annotations[toolchainv1alpha1.UserSignupActivationCounterAnnotationKey]
+
 	// Check if the UserSignup is waiting for phone verification to be finished
-	// Check also if the status doesn't contain the approved condition set to true - it there is, then it means that it's reactivated UserSignup and we should skip it here
-	if states.VerificationRequired(instance) && !condition.IsTrue(instance.Status.Conditions, toolchainv1alpha1.UserSignupApproved) {
+	// Check also if the UserSignup doesn't have at least one activation - if it has, then it means that it's reactivated UserSignup and we should skip it here
+	if states.VerificationRequired(instance) && (!activationsAnnotationPresent || activations == "0") {
```

Comment:
IIRC, we never set the annotation value to `0` (but it's fine/safe to keep this assertion)

Reply:
ah, right - I got confused with this code https://github.com/codeready-toolchain/host-operator/blob/master/controllers/usersignup/usersignup_controller.go#L420-L428
at the first glance, it's not really clear that the update for `"0"` is skipped